internal/backends/nodejs/grab.go

```diff
@@ -77,168 +75,104 @@ func parseFile(contents []byte, results chan parseResult) {
  (#eq? @function "require"))
 `
 
-	query, err := sitter.NewQuery([]byte(importsQuery), language)
+	js := javascript.GetLanguage()
+	jsPkgs, err := util.GuessWithTreeSitter(dir, js, importsQuery, jsPathGlobs, []string{})
 	if err != nil {
-		// This should never happen - the query is hardcoded
-		log.Fatalln(err, importsQuery)
+		util.Die("couldn't guess imports: %s", err)
 	}
 
-	qc := sitter.NewQueryCursor()
+	ts := typescript.GetLanguage()
+	tsPkgs, err := util.GuessWithTreeSitter(dir, ts, importsQuery, tsPathGlobs, []string{})
+	if err != nil {
+		util.Die("couldn't guess imports: %s", err)
+	}
 
-	node, err := sitter.ParseCtx(context.Background(), contents, language)
+	tsx := tsx.GetLanguage()
+	tsxPkgs, err := util.GuessWithTreeSitter(dir, tsx, importsQuery, tsxPathGlobs, []string{})
 	if err != nil {
-		results <- parseResult{nil, false}
-		return
+		util.Die("couldn't guess imports: %s", err)
 	}
 
-	qc.Exec(query, node)
+	modules := append([]string{}, jsPkgs...)
+	modules = append(modules, tsPkgs...)
+	modules = append(modules, tsxPkgs...)
```

Comment:
Is it useful to enforce uniqueness in the modules list here to avoid doing extra work later on?

Reply:
done in [99bb975](https://github.com/replit/upm/pull/154/commits/99bb975f8a394b465a74e040c635d1c219128abb)